packages/core/src/components/Timer.tsx

```diff
@@ -0,0 +1,139 @@
+import React, {
+  useState,
+  useEffect,
+  useRef,
+  useImperativeHandle,
+  forwardRef,
+} from "react";
+import { Text, StyleSheet, TextStyle } from "react-native";
+
+interface TimerProps {
+  style?: TextStyle;
+  initialTime?: number;
+  updateInterval?: number;
+  format?: "ss" | "mm:ss" | "hh:mm:ss" | "ss:ms" | "mm:ss:ms" | "hh:mm:ss:ms";
+  onTimerChange?: (time: number) => void;
+  onTimerEnd?: () => void;
+  countDirection?: "up" | "down";
+}
+
+export interface TimerHandle {
+  start: () => void;
+  stop: () => void;
+  reset: (newTime?: number) => void;
+}
+
+const Timer = forwardRef<TimerHandle, TimerProps>(
+  (
+    {
+      style,
+      initialTime,
+      updateInterval = 1000,
+      format = "mm:ss",
+      onTimerChange,
+      onTimerEnd,
+      countDirection = "up",
+    },
+    ref
+  ) => {
+    const defaultInitialTime = countDirection === "up" ? 0 : 100000;
+    const [time, setTime] = useState(initialTime ?? defaultInitialTime);
+    const timerRef = useRef<NodeJS.Timeout | null>(null);
```

Comment:
This timer is never cleared on component unmount, which can cause a leak where we have timers running that do not belong to any component. I think we should use a `useEffect` somewhere with the cleanup callback. Something like this https://stackoverflow.com/a/63334017

Reply:
Thanks, good point